lib/web_ui/lib/src/engine/pointer_binding.dart

```diff
@@ -244,103 +229,139 @@ abstract class BaseAdapter {
   void _addWheelEventListener(html.EventListener handler) {
     final dynamic eventOptions = js_util.newObject();
     final html.EventListener jsHandler = js.allowInterop((html.Event event) => handler(event));
-    _nativeListeners['wheel'] = jsHandler;
+    _BaseAdapter._nativeListeners['wheel'] = jsHandler;
     js_util.setProperty(eventOptions, 'passive', false);
     js_util.callMethod(
-      domRenderer.glassPaneElement,
+      glassPaneElement,
       'addEventListener', <dynamic>[
         'wheel',
         jsHandler,
         eventOptions
       ]
     );
-
   }
 }
 
-const int _kPrimaryMouseButton = 0x1;
-const int _kSecondaryMouseButton = 0x2;
-
-int _pointerButtonFromHtmlEvent(html.Event event) {
-  if (event is html.PointerEvent) {
-    final html.PointerEvent pointerEvent = event;
-    return pointerEvent.button == 2
-        ? _kSecondaryMouseButton
-        : _kPrimaryMouseButton;
-  } else if (event is html.MouseEvent) {
-    final html.MouseEvent mouseEvent = event;
-    return mouseEvent.button == 2
-        ? _kSecondaryMouseButton
-        : _kPrimaryMouseButton;
+typedef void _PointerStateDataCallback({
+  @required ui.PointerChange change,
+  @required int buttons,
+});
+
+class _PointerStateManager {
+  int _pressedButtons = 0;
+
+  // Transform html.PointerEvent.buttons to Flutter's PointerEvent buttons.
+  int _htmlButtonsToFlutterButtons(int buttons) {
+    // Flutter's button definition conveniently matches that of JavaScript
+    // from primary button (0x1) to forward button (0x10), which allows us to
+    // avoid transforming it bit by bit.
+    return buttons & _kButtonsMask;
+  }
+
+  void handleDown({@required _PointerStateDataCallback dataCallback, @required int buttons}) {
+    // TODO(flutter_web): Remove this temporary fix for right click
+    // on web platform once context guesture is implemented.
+    if (_pressedButtons != 0) {
+      _pressedButtons = 0;
+      dataCallback(change: ui.PointerChange.up, buttons: _pressedButtons);
```

Comment:
The "pointerdown" events are dispatched only when the pointer changes from 0 buttons pressed to any buttons, while "pointerup" events are any buttons to 0 buttons. Button changes from some buttons to another set of buttons is dispatched as "pointermove" event.

Reply:
Using a mouse, both the left button and the right button can be down at the same time.


```
   [1]           [2]          [3]          [4]
down(left) -> down(right) -> up(right) -> up(left)
```
In the scenario above, I think [2] and [4] won't be sent to Flutter. Is my understanding correct?

Reply:
It's not correct. In your scenario, JavaScript will send the following events:
```
       [1]                    [2]                   [3]                   [4]
down(buttons: 0x1) -> move(buttons: 0x3) -> move(buttons: 0x1) -> up(buttons: 0x0)
```

Since Flutter always expect all and continuous events, Flutter will expect similar events:
```
       [1]                    [2]                   [3]                   [4]
Down(buttons: 0x1) -> Move(buttons: 0x3) -> Move(buttons: 0x1) -> Up(buttons: 0x0)
```

Reply:
Wow this is surprising. Nice finding.

Reply:
I've just found today that the way I described applies to `PointerEvent`, while `MouseEvent` goes with the way you described. #HtmlStandard

Reply:
I wrote this in the document of `_MouseAdapter`.

Reply:
lol welcome to the web :)